pkgs/applications/networking/irc/url-bot-rs/default.nix

```diff
@@ -0,0 +1,45 @@
+{ stdenv
+, lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, openssl
+, sqlite
+, libiconv
+, Security
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "url-bot-rs";
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "nuxeh";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0y183z86rkpxxvvzrkkpdxxxpd7lr8nnmwi5nv7k8cfb8fa0xl8n";
+  };
+
+  cargoSha256 = "0081liwxwxd86amlpkwncdnbbykab42c2myqsxr44k6a9yliivsp";
```

Comment:
My 2cents: don't. Add the Cargo.lock next to this file and set the `cargoLock = ./Cargo.lock` attribute instead. No need to fight with FODs.